cinder/tests/unit/volume/drivers/lightos/test_lightos_storage.py

```diff
@@ -0,0 +1,285 @@
+            "10.10.10.71,10.10.10.72,10.10.10.73"
+        configuration.lightos_api_port = 443
+        configuration.lightos_jwt = None
+        configuration.lightos_snapshotname_prefix = 'openstack_'
+        configuration.lightos_intermediate_snapshot_name_prefix = 'for_clone_'
+        configuration.lightos_default_compression_enabled = False
+        configuration.lightos_default_num_replicas = 3
+        configuration.num_volume_device_scan_tries = \
+            constants.DEVICE_SCAN_ATTEMPTS_DEFAULT
+        configuration.lightos_api_service_timeout = \
+            constants.LIGHTOS_API_SERVICE_TIMEOUT
+        configuration.driver_ssl_cert_verify = False
+        # for some reason this value is not initialized by the driver parent
+        # configs
+        configuration.volume_name_template = 'volume-%s'
+
+        configuration.initiator_connector = \
+            "cinder.tests.unit.volume.drivers.lightos." \
+                "test_lightos_storage.InitiatorConnectorMock"
+
+        self.driver = lightos.LightOSVolumeDriver(configuration=configuration)
+
+        self.db: Dict[str, str] = {
+            "projects": {
+                "default": {
+                    "volumes": [],
+                    "snapshots": []
+                }
+            }
+        }
+
+        # define a default send_cmd override to return default values.
+        def send_cmd_default_mock(cmd, timeout, **kwargs):
+            if cmd == "get_nodes":
+                return (httpstatus.OK, FAKE_LIGHTOS_CLUSTER_NODES)
+            if cmd == "get_node":
+                self.assertTrue(kwargs["UUID"])
+                for node in FAKE_LIGHTOS_CLUSTER_NODES["nodes"]:
+                    if kwargs["UUID"] == node["UUID"]:
+                        return (httpstatus.OK, node)
+                return (httpstatus.NOT_FOUND, node)
+            elif cmd == "get_cluster_info":
+                return (httpstatus.OK, FAKE_LIGHTOS_CLUSTER_INFO)
+            elif cmd == "create_volume":
+                project_name = kwargs["project_name"]
+                volume = {
+                    "project_name": project_name,
+                    "name": kwargs["name"],
+                    "size": kwargs["size"],
+                    "n_replicas": kwargs["n_replicas"],
+                    "compression": kwargs["compression"],
+                    "src_snapshot_name": kwargs["src_snapshot_name"],
+                    "acl": kwargs["acl"],
+                    "UUID": str(uuid.uuid4()),
+                    "state": "Available",
+                }
+                project = self.db["projects"].setdefault(project_name, {})
+                project["volumes"].append(volume)
+                return (httpstatus.OK, volume)
+            elif cmd == "delete_volume":
+                project_name = kwargs["project_name"]
+                volume_uuid = kwargs["volume_uuid"]
+                project = self.db["projects"].get(project_name, None)
+                if project:
```

Comment:
if we dont find project we seem to fall down to raise runtime error.
i think close emulation of product is
 return (httpstatus.INVALID_ARGUMENT, None)